extensions/azurecore/src/account-provider/auths/azureAuth.ts

```diff
@@ -354,36 +462,64 @@ export abstract class AzureAuth implements vscode.Disposable {
 	}
 
 
+	public async getTenantsMsal(token: string): Promise<Tenant[]> {
+		const tenantUri = url.resolve(this.metadata.settings.armResource.endpoint, 'tenants?api-version=2019-11-01');
+		try {
+			Logger.verbose('Fetching tenants', tenantUri);
+			let tenantList: string[] = [];
+			const tenantResponse = await this.makeGetRequest(tenantUri, token);
+			const tenants: Tenant[] = tenantResponse.data.value.map((tenantInfo: TenantResponse) => {
+				if (tenantInfo.displayName) {
```

Comment:
should we log the else case here to help diagnosing errors?  Or it's ok to return from this method without updating the tenant list?  At a higher point, please look through all the failure cases in the PR and ensure we're logging all the relevant info to help try to determine what's going on when auth fails.

Reply:
..from following return it looks like this case is assumed to be "work or school" account.  I guess the comment is just confirming this is true and mentioning to log all the error cases.

Reply:
I don't quite understand why missing display name for tenant would be replaced with 'Work or School Account', since there can be multiple tenants under a Work or School accounts, it shouldn't be named as a tenant.